pkg/controller/storagecluster/reconcile.go

```diff
@@ -550,3 +574,51 @@ func newStorageClassDeviceSets(storageDeviceSets []ocsv1.StorageDeviceSet, topol
 
 	return storageClassDeviceSets
 }
+
+func (r *ReconcileStorageCluster) isActiveStorageCluster(instance *ocsv1.StorageCluster) (bool, error) {
+	storageClusterList := ocsv1.StorageClusterList{}
+	opts := &client.ListOptions{
+		Namespace: instance.Namespace,
+		Raw: &metav1.ListOptions{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       instance.Kind,
+				APIVersion: instance.APIVersion,
+			},
+		},
+	}
+
+	// instance is already marked for deletion
+	// do not mark it as active
+	if instance.Status.Phase == "" &&
+		!instance.GetDeletionTimestamp().IsZero() {
+		return false, nil
+	}
+
+	err := r.client.List(context.TODO(), opts, &storageClusterList)
+	if err != nil {
+		return false, fmt.Errorf("Error fetching StorageClusterList. %+v", err)
+	}
+
+	// There is no StorageCluster
+	if storageClusterList.Items == nil {
+		return true, nil
+	}
+	// There are many StorageClusters. Check if this is Active
+	for _, storageCluster := range storageClusterList.Items {
+		if storageCluster.Status.Phase != statusutil.PhaseIgnored &&
+			storageCluster.ObjectMeta.Name != instance.ObjectMeta.Name {
+			// Both StorageClusters are in creation phase
+			// Tiebreak using CreationTimestamp and Alphanumeric ordering
+			if storageCluster.Status.Phase == "" {
+				if storageCluster.CreationTimestamp.Before(&instance.CreationTimestamp) {
+					return false, nil
+				} else if storageCluster.CreationTimestamp.Equal(&instance.CreationTimestamp) && storageCluster.Name < instance.Name {
+					return false, nil
+				}
+				return true, nil
```

Comment:
we aren't guaranteed just because this instance wins the tie breaker that it's the active instance. There could be more storageClusters in the list.

Reply:
Added a check to `return true` only if we are at the last item.